### DISCUSSION TITLE
long inference latency for speechbrain/spkrec-ecapa-voxceleb

### DISCUSSION THREAD

**Discussion by cirenehc** [speechbrain/speechbrain#1666] [Q&A]
I've been doing some empirical latency measurements for the pretrained  Speaker Recognition VoxCeleb model (speechbrain/spkrec-ecapa-voxceleb). I'm getting a ~8s inference latency on a 4s audio for [verify_batch](https://speechbrain.readthedocs.io/en/latest/_modules/speechbrain/pretrained/interfaces.html#SpeakerRecognition.verify_batch)) on a machine with 2.6 GHz 6-Core Intel Core i7. Are there ways to speed it up? In my production set up I'm going to be doing inferencing on a raspberry pi so the hardware limitation is even greater.

**Comment by dragen1860** [comment]
@cirenehc it's abnomal. I tested the inference time with onnx on mobile phone. it's takes ~90ms on 865 cpu for ~2s segment. it's would be extremely faster if you run on x86 desktop cpu. add me wechat: `dragen1860` if you have further problems.
